qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -0,0 +1,273 @@
+
+        if isinstance(inst, QuantumCircuit) and inst.global_phase != 0:
+            aer_state.apply_global_phase(inst.global_phase)
+
+        AerStatevector._aer_evolve_circuit(aer_state, circuit, range(num_qubits))
+
+        return aer_state.move_to_ndarray(), aer_state
+
+    @classmethod
+    def _aer_evolve_circuit(cls, aer_state, circuit, qubits):
+        """Apply circuit into aer_state"""
+        for inst, qargs, _ in circuit.data:
+            AerStatevector._aer_evolve_instruction(aer_state, inst,
+                                                   [qubits[circuit.find_bit(qarg).index]
+                                                    for qarg in qargs])
+
+    @classmethod
+    def _aer_evolve_instruction(cls, aer_state, inst, qubits):
+        """Apply instruction into aer_state"""
+        params = inst.params
+        if inst.name == 'u':
+            aer_state.apply_mcu(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1],
+                                params[0], params[1], params[2])
+        elif inst.name in ['x', 'cx', 'ccx']:
+            aer_state.apply_mcx(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
+        elif inst.name in ['y', 'cy']:
+            aer_state.apply_mcy(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
+        elif inst.name in ['z', 'cz']:
+            aer_state.apply_mcz(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
+        elif inst.name == 'unitary':
+            aer_state.apply_unitary(qubits, inst.params[0])
+        elif inst.name == 'diagonal':
+            aer_state.apply_diagonal(qubits, inst.params[0])
+        else:
+            definition = inst.definition
+            if definition is inst:
+                raise AerError('cannot decompose ' + inst.name)
+            AerStatevector._aer_evolve_circuit(aer_state, definition, qubits)
+
+    @classmethod
+    def from_label(cls, label):
+        """Return a tensor product of Pauli X,Y,Z eigenstates.
+
+        .. list-table:: Single-qubit state labels
+           :header-rows: 1
+
+           * - Label
+             - Statevector
+           * - ``"0"``
+             - :math:`[1, 0]`
+           * - ``"1"``
+             - :math:`[0, 1]`
+           * - ``"+"``
+             - :math:`[1 / \\sqrt{2},  1 / \\sqrt{2}]`
+           * - ``"-"``
+             - :math:`[1 / \\sqrt{2},  -1 / \\sqrt{2}]`
+           * - ``"r"``
+             - :math:`[1 / \\sqrt{2},  i / \\sqrt{2}]`
+           * - ``"l"``
+             - :math:`[1 / \\sqrt{2},  -i / \\sqrt{2}]`
+
+        Args:
+            label (string): a eigenstate string ket label (see table for
+                            allowed values).
+
+        Returns:
+            AerStatevector: The N-qubit basis state density matrix.
+
+        Raises:
+            QiskitError: if the label contains invalid characters, or the
+                         length of the label is larger than an explicitly
+                         specified num_qubits.
+        """
```

Comment:
(I think) you can omit the docstring if you just want to inherit it from the parent class.